docs: add metadata to all remaining pages (#1246)
canonical/rockcraft: docs/explanation/overlay-step.rst

.. meta::

>     :description: Explanation of the Rockcraft overlay step, a unique build phase that modifies the base filesystem before the build step using packages, scripts, and overlays.

Reviewer: ```suggestion
    :description: Explanation of the overlay step in the parts lifecycle, which modifies the base filesystem before the build step begins.
```

- I think we decided not to use "phase" to describe the lifecycle steps, but I could be misremembering.
- Let's not compare Rockcraft to other crafts in the descriptions ("unique"). It might confuse SEO.

Author: I was not aware, thanks for letting me know!
